core/services/terraform_service.rb

```diff
@@ -119,6 +119,22 @@ def self.nodes_to_resources(nodes, resource_type)
     nodes.map { |node| [node, "#{resource_type}.#{node}"] }.to_h
   end
 
+  # Searches additional disk resources declared in the terraform config file
+  #
+  # @param path [String] path to the directory with terraform configuration
+  # @return [Hash] Hash with node name keys and resources names values
+  # in format 'aws_volume_attachment.node-disk-attachment'
+  def self.additional_disk_resources(path)
+    configuration_path = "#{path}/#{TerraformConfigurationGenerator::CONFIGURATION_FILE_NAME}"
```

Comment:
For each node that has the attached disk, there is a "NAME-disk-attachment" resource. Why do we need to parse the terraform configuration file in order to determine that? We can determine that requirement solely from the template file.

Another point - why do you tie this logic to the AWS only? It only depends on the requirement of the disk attachment.